azext_edge/edge/providers/rpsaas/base_provider.py

```diff
@@ -0,0 +1,87 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from typing import Any, Dict, List, Optional
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class RPSaaSBaseProvider:
+    def __init__(
+        self,
+        cmd,
+        api_version: str,
+        resource_type: str,
+        **kwargs
+    ):
+        from azure.cli.core.commands.client_factory import get_subscription_id
+        from ...util.az_client import get_resource_client
+
+        self.cmd = cmd
+        self.subscription = get_subscription_id(cmd.cli_ctx)
+        self.resource_client = get_resource_client(subscription_id=self.subscription)
+        self.api_version = api_version
+        self.resource_type = resource_type
+        self.required_extension = kwargs.get("required_extension")
```

Comment:
required_extension should probably be required. There are direct references in ADRBaseProvider

Reply:
This is the base class of the ADRBaseProvider - other services may not need this.

I made sure that the attribute will be none if there aren't any required extensions. That way you can be sure that this attribute will exist within the object.